src/tests/nativeaot/SmokeTests/HelloWasm/HelloWasm.cs

```diff
@@ -2456,6 +2462,456 @@ private static void TestFilterNested()
+            }
+        }
+
+        StartTest("TestDynamicStackAlloc(double release on return)");
+        {
+            bool doReturnWithEH = false;
+            DoAlloc(out byte* addrOne);
+            DoDoubleAlloc(&doReturnWithEH);
+            DoAlloc(out byte* addrTwo);
+            result = addrOne == addrTwo;
+        }
+        EndTest(result);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static void DoAllocAndThrow(out byte* addr)
+        {
+            byte* stk = stackalloc byte[StkAllocSize];
+            addr = stk;
+
+            try
+            {
+                Volatile.Write(ref stk[StkAllocSize - 1], 1);
+                Volatile.Write(ref *(int*)null, 0);
+            }
+            catch (DivideByZeroException)
+            {
+                Volatile.Read(ref stk[StkAllocSize - 1]);
+            }
+        }
+
+        StartTest("TestDynamicStackAlloc(release on EH return)");
+        {
+            byte stkByte;
+            byte* addrOne = null;
+            byte* addrTwo = &stkByte;
+            try
+            {
+                DoAllocAndThrow(out addrOne);
+            }
+            catch (NullReferenceException)
+            {
+            }
+            try
+            {
+                DoAllocAndThrow(out addrTwo);
+            }
+            catch (NullReferenceException)
+            {
+            }
+
+            result = addrOne == addrTwo;
+        }
+        EndTest(result);
+
+        StartTest("TestDynamicStackAlloc(double release on EH return)");
+        {
+            DoAlloc(out byte* addrOne);
+            try
+            {
+                bool doReturnWithEH = true;
+                DoDoubleAlloc(&doReturnWithEH);
+            }
+            catch (NullReferenceException)
+            {
+            }
+            DoAlloc(out byte* addrTwo);
+
+            result = addrOne == addrTwo;
+        }
+        EndTest(result);
+
+        StartTest("TestDynamicStackAlloc(release on EH return does not corrupt live state)");
+        {
+            byte* stkOne = stackalloc byte[StkAllocSize];
+            Volatile.Write(ref stkOne[StkAllocSize - 1], 2);
+
+            byte* stkTwo = null;
+            byte* stkThree = null;
+            try
+            {
+                DoAllocAndThrow(out stkTwo);
+            }
+            catch (NullReferenceException)
+            {
+                Volatile.Read(ref stkOne[StkAllocSize - 1]);
+            }
+
+            try
+            {
+                DoAlloc(out stkThree);
+                Volatile.Write(ref stkThree[StkAllocSize - 1], 10);
+
+                result = stkTwo == stkThree && stkOne != stkThree && Volatile.Read(ref stkOne[StkAllocSize - 1]) == 2;
+                Volatile.Write(ref *(int*)null, 0);
+            }
+            catch (NullReferenceException)
+            {
+                Volatile.Read(ref stkThree[StkAllocSize - 1]);
+            }
+        }
+        EndTest(result);
```

Comment:
Should we be resetting `result` to false before each test, in case the code never gets to the assignment?

Reply:
Did that in cases where `result` assignment wasn't the last a test would do.